src/bci_build/package/__init__.py

```diff
@@ -392,11 +392,13 @@ def build_name(self) -> str | None:
 
     @property
     def build_version(self) -> str | None:
+        epoch = f"{self.os_epoch}." if self.os_epoch else ""
         if self.os_version.is_sle15:
-            epoch = ""
-            if self.os_epoch:
-                epoch = f"{self.os_epoch}."
             return f"15.{epoch}{int(self.os_version.value)}"
+        if self.os_version.is_slfo:
+            return (
+                f"16.{epoch}{str(float(self.os_version.value) % 1).partition('.')[2]}"
+            )
```

Comment:
This will lead to funny rounding errors if we ever bump `os_version` to `16.1`:
```
>>> str(16.1 % 1)
'0.10000000000000142'
```

How about omitting the modulo call then? You're throwing the major version away anyway
```suggestion
        if self.os_version.is_slfo:
            return (
                f"16.{epoch}{str(float(self.os_version.value)).partition('.')[2]}"
            )
```

Reply:
as you wish to remove epoch handling anyway, we don't need to make it that convoluted.